src/main/java/core/basesyntax/lib/Injector.java

```diff
@@ -0,0 +1,30 @@
+package core.basesyntax.lib;
+
+import core.basesyntax.dao.BetDao;
+import core.basesyntax.dao.UserDao;
+import core.basesyntax.facotry.Factory;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+
+public class Injector {
+    public static Object getInstance(Class clazz) throws NoSuchMethodException, IllegalAccessException, InvocationTargetException, InstantiationException {
+        Constructor constructor = clazz.getDeclaredConstructor();
+        Object instance = constructor.newInstance();
+
+        Field[] fields = clazz.getDeclaredFields();
+        for (Field field : fields) {
+            if (field.getAnnotation(Inject.class) != null) {
+                field.setAccessible(true);
+                if (field.getType().equals(UserDao.class)) {
+                    field.set(instance, Factory.getUserDao());
+                }
+                if (field.getType().equals(BetDao.class)) {
+                    field.set(instance, Factory.getBetDao());
```

Comment:
We need to check have class UserDaoImpl or BetDaoImpl annotation @dao before set the instance

Reply:
Also we need to create custom exception for throw it when we can not found the annotation